src/infra/core/src/query_service_impl.rs

```diff
@@ -73,6 +75,18 @@ impl QueryServiceImpl {
         // See: https://github.com/apache/datafusion/issues/13504
         cfg.options_mut().execution.parquet.schema_force_view_types = false;
 
+        if let Some(overrides) = session_config_overrides {
+            if let Some(val) = overrides.target_partitions {
+                cfg.options_mut().execution.target_partitions = val;
+            }
+            if let Some(val) = overrides.minimum_parallel_output_files {
+                cfg.options_mut().execution.minimum_parallel_output_files = val;
+            }
+            if let Some(val) = overrides.soft_max_rows_per_output_file {
+                cfg.options_mut().execution.soft_max_rows_per_output_file = val;
```

Comment:
Is there no way to override this in `write_parquet`? 

I would like to avoid tying `QueryService` interface so closely to DataFusion internals by surfacing such in-depth config parameters.

Reply:
I didn't find options.
This is where it's being used: https://github.com/apache/datafusion/blob/2464703c84c400a09cc59277018813f0e797bb4e/datafusion/core/src/datasource/file_format/write/demux.rs#L150-L150
And TaskContext (SessionContext.state) comes from `write_parquet`, where new dataframe is being created: https://github.com/apache/datafusion/blob/2464703c84c400a09cc59277018813f0e797bb4e/datafusion/core/src/dataframe/parquet.rs#L85-L85
And latter comes from context we create.

If you want to have these services completely independent, we can just have session creation code (mostly)dup in new service, or move `create_session` and `create_session_with_config_overrides` to a third service and have it as a dependency for QueryService and ExportService.

Reply:
My prev statement is not quite true. I figured out how to reassemble DataFrame with hacked session config.